lib/sles4sap/sap_deployment_automation_framework/deployment.pm

```diff
@@ -562,8 +563,29 @@ sub prepare_sdaf_project {
     assert_script_run("cd $deployment_dir");
     assert_script_run('mkdir -p ' . log_dir());
 
+    # Calculate SDAF version used for deployment and picks latest -1
+    # SDAF_GIT_AUTOMATION_BRANCH variable will override calculated value
+    my $branch = get_var('SDAF_GIT_AUTOMATION_BRANCH', '');
+    if (!$branch || $branch eq 'latest') {
+        my $tags = script_output("curl -s https://api.github.com/repos/Azure/sap-automation/tags | jq -r '.[].name' | sort -rV");
+        record_info("Releases: $tags");
+        my @releases = split('\n', $tags);
+        my $branch_expected = $releases[1];
```

Comment:
do you like to try this: 
`my $branch_expected = ($branch eq 'latest') ? $releases[0] : $releases[1];`